### PR TITLE
Add release instructions

### PR DESCRIPTION
**Pull request by adleong**
Add instructions for releasing Linkerd

Signed-off-by: Alex Leong <alex@buoyant.io>

### CODE REVIEW

**Comment by Pothulapati** [comment]
Hmm, I was following the Wiki until now https://github.com/linkerd/linkerd2/wiki/Linkerd's-Release-Process

Wondering if we should remove the Wiki and keep this? 🤔

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+That will kick off a CI Release workflow run that will:
+
+- Build and push the docker images for the tag that was created
+- Run the k3d integration tests in the Github actions VMs themselves
+- Run a k3d integration test on a separate ARM64 host
+- Create a release in Github, and upload the CLI binaries with their checksums
+- Dispatch an event caught by the website repo that triggers a website rebuild
+  which will update the edge/stable versions in the website
+- Retrive the installation script from [run.linkerd.io](https://run.linkerd.io)
```
Comment (alpeb) [inline RELEASE.md:253]:
```suggestion
- Retrieve the installation script from [run.linkerd.io](https://run.linkerd.io)
```

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+- [linkerd-dev](https://lists.cncf.io/g/cncf-linkerd-dev/join)
+
+Include the full release notes in the email. Liberally apply emoji. ⭐
+
+## 13. Send an announcement to Linkerd Slack's #announcements channel
+
+Ensure that you send a brief summary of the release in Linkerd Slack's
+[#announcement](https://linkerd.slack.com/messages/C0JV5E7BR) channel.
+
+## 15. Add a community page announcement to the website repo
```
Comment (alpeb) [inline RELEASE.md:299]:
```suggestion
## 14. Add a community page announcement to the website repo
```

**Review by alpeb — APPROVED** [review]
:+1: 
We should definitely delete the release instructions in the wiki, which are stale. And also remove the buoyant-internal one of course.

File: RELEASE.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+Almost all the charts are always updated, at least because the docker image
+versions referred in their templates change with each release. One exception is
+`linkerd-crds`, which doesn't contain image references. So this chart only
+requires bumping its `version` if there were changes in its templates files.
+
+## 7. Update the release notes
+
+On this branch, add the release notes for this version in `CHANGES.md`. Make
+sure it follows the style outlined in [the Release Notes Style
```
Comment (mateiidavid) [inline RELEASE.md:122]: Think `style-guide.md` doesn't exist in this repo, should it be taken out?

**Review by mateiidavid — APPROVED** [review]

**Review by kleimkuhler — APPROVED** [review]

**Pull request merged**